### PR TITLE
Fix RawTable::allocation_info for empty table

### LINKED ISSUE #376: `RawTable::allocation_info` returns non zero size for a fresh empty map
The following code:

```
#[test]
fn test() {
    assert_eq!(hashbrown::raw::RawTable::<()>::new().allocation_info().1.size(), 0);
}
```

fails with:

```
thread 'test' panicked at 'assertion failed: `(left == right)`
  left: `17`,
 right: `0`', src/lib.rs:3:5
```

Since `RawTable::new` doesn't allocate any memory I'd expect this to return 0.

### PR DESCRIPTION
**Pull request by stepancheg**
Fixes #376

### CODE REVIEW

File: src/raw/mod.rs [hunk 1]
```diff
@@ -531,7 +531,12 @@ impl<T, A: Allocator + Clone> RawTable<T, A> {
     #[inline]
     #[cfg(feature = "raw")]
     pub fn allocation_info(&self) -> (NonNull<u8>, Layout) {
-        self.table.allocation_info(Self::TABLE_LAYOUT)
```
Comment (Amanieu) [inline src/raw/mod.rs:534]: Can this check be moved into the method in `RawTableInner`?

Comment (stepancheg) [reply src/raw/mod.rs:534]: Are you sure? This way we will do branch (albeit trivial) for paths we don't use in normal operation.

At the moment I added assertion to `RawTableInner.allocation_info`.

Comment (Amanieu) [reply src/raw/mod.rs:534]: The reason for the split between `RawTable` and `RawTableInner` is to reduce the amount of LLVM IR generated and improve compilation times. In release builds, all the `RawTableInner` methods are inlined into the `RawTable` methods.

Comment (stepancheg) [reply src/raw/mod.rs:534]: Got it. Updated the PR.

**Pull request merged**